lib/kubernetes-deploy/pod_logs.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class PodLogs
+    def initialize(logger:, pod_name:, container_names:)
+      @logger = logger
+      @pod_name = pod_name
+      @container_names = container_names
+      @last_timestamp_seen_for_container = {}
+      @container_logs = Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def sync(kubectl)
+      latest_logs = fetch_logs(kubectl, since: @last_timestamp_seen_for_container.values.min)
+      latest_logs.each do |container_name, log_lines|
+        next if log_lines.empty?
+
+        log_lines.each do |line|
+          timestamp, msg = split_timestamped_line(line)
+          next if likely_duplicate?(timestamp, container_name)
+          @container_logs[container_name] << msg
+          @last_timestamp_seen_for_container[container_name] = timestamp
+        end
+      end
+    end
+
+    def print_latest
+      @last_printed_indexes ||= Hash.new { |hash, key| hash[key] = -1 }
+      @container_logs.each do |container_name, logs|
+        prefix = "[#{container_name}]  " if @container_names.length > 1
```

Comment:
Part of me was thinking that we have to do some ugly things because a pod can have multiple containers. But is that something we actually need/want to allow for the runner?

Reply:
We definitely need to not care if they have extra containers, but they do already have to tell us which one is doing the main work (so we can inject the env vars/args/etc in that one). Maybe we could set `pod.main_container` or something and pick that one. If we use this in deploys though (which I'd like to), we won't have that information, so we'd probably need to keep the prefixing for that case. WDYT? 🤔